contracts/voting/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+#![no_std]
+
+use soroban_sdk::{
+    contracterror, contractimpl, contracttype, vec, Address, ConversionError, Env, Map, Symbol,
+};
+
+#[contracterror]
+#[derive(Clone, Debug, Copy, Eq, PartialEq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum Error {
+    Conversion = 1,
+    KeyExpected = 2,
+    NotFound = 3,
+    AlreadyVoted = 4,
+    DuplicatedEntity = 5,
+    Overflow = 6,
+    VotingClosed = 7,
+    NotValidID = 8,
+}
+
+impl From<ConversionError> for Error {
+    fn from(_: ConversionError) -> Self {
+        Error::Conversion
+    }
+}
+
+#[contracttype]
+#[derive(Clone, Copy)]
+pub enum DataKey {
+    Admin = 0,
+    VoterList = 1,
+    Proposals = 2,
+}
+
+pub struct ProposalVotingContract;
+
+#[contractimpl]
+impl ProposalVotingContract {
+    pub fn init(env: Env, admin: Address) {
+        env.storage().set(&DataKey::Admin, &admin);
+        env.storage()
+            .set(&DataKey::Proposals, &Map::<u64, Proposal>::new(&env));
+    }
+
+    pub fn create_proposal(env: Env, id: u64) -> Result<(), Error> {
+        let voting_period_secs = 3600; // one hour
+        let target_approval_rate_bps = 50_00; // At least 50% of the votes to be approved.
+        let total_voters = 1000; // Up to 1000 participants.
```

Comment:
I think this parameters could be easily configured in the Wizard.

Reply:
We can move them to the init. Lets just take into account if we move them there, we need to store them in the ledger for later retrieval in the `create_proposal` function. Like we are doing [here](https://github.com/eigerco/nebula/pull/13#discussion_r1258918729).

I have doubts regarding this, because adding/retrieving elements from the storage, has consequences in fees. [See this docs](https://soroban.stellar.org/docs/fundamentals-and-concepts/fees-and-metering#resource-fee). So maybe they are just good as hardcoded values ? I really dont know. 

I will proceed on moving them to init in the meanwhile :+1:

Reply:
You are right. Maybe add a comment in the code for future consideration of making them hardcoded if the fee is too high :)

Reply:
Fixed at e3cf2f7e185e81085eb3f3583092b0e902ee7e40